demos/free_surface/free_surface.py

```diff
@@ -0,0 +1,355 @@
+boring_cmap = plt.get_cmap("coolwarm", 25)
+
+for i in range(len(reader.time_values)):
+    reader.set_active_time_point(i)
+    data = reader.read()[0]
+
+    # Artificially warp the output data in the vertical direction by the free surface height
+    warped = data.warp_by_scalar(scalars='eta', normal=(0, 1, 0), factor=1.5)
+
+    # Add the warped temperature field to the frame
+    plotter.add_mesh(
+        warped,
+        scalars="Temperature",
+        lighting=False,
+        show_edges=False,
+        clim=[0, 1],
+        cmap=boring_cmap,
+        scalar_bar_args={'position_x': 0.2, 'position_y': 0.05}
+    )
+    arrows = data.glyph(orient="Velocity", scale="Velocity", factor=0.001, tolerance=0.05)
+    plotter.add_mesh(arrows, color="black")
+    # Centre camera on xy plane and write frame
+    plotter.camera_position = "xy"
+    plotter.write_frame()
+    plotter.clear()
+
+# Closes and finalizes movie
+plotter.close()
+
+
+# -
+
+# ![SegmentLocal](temperature_warp.gif "segment")
+
+# As before with the base case, the buoyancy differences due to the starting temperature configuration drives an overturning flow. You should see that the temperature field at the final time is very similar to the base case result - in fact they are the same!
+#
+# Given this problem reaches a steady state adding a free surface should not change that behaviour, it will just change how the simulation reaches that steady state. We can see that the free surface has been pushed up above regions of hotter mantle, where thermal convection is driving positive vertical flow and is sinking where there is colder mantle below, which all makes intuitive sense.
+#
+#
+
+# We can even verify that the stress exerted by the deviation of the free surface from the reference state is the same as the when calculating dynamic topography from a free-slip simulation. As a recap the free surface normal boundary condition is
+#
+# \begin{equation}
+#       \textbf{n} \cdot \boldsymbol{\sigma }' \cdot \textbf{n}  = - \Delta\rho_{FS} g \eta.
+# \end{equation}
+#
+# In 2D, on the top surface, this simplifies to
+# \begin{equation}
+#      -p'  + 2\mu \frac{\partial v}{\partial y }  = - \Delta\rho_{FS} g \eta.
+# \end{equation}
+#
+# For a nondimensional simulation this simplifies to
+# \begin{equation}
+#      -p_{\star}'  + 2 \frac{\partial v_{\star}}{\partial y_{\star} }  = - Ra_{FS} \eta_{\star}+ Ra T_{\star}   \eta_{\star},
+# \end{equation}
+# where $\star$ indicates a nondimensional variable. Rearranging, our expected dynamic topography at steady state would be
+# \begin{equation}
+#      \eta_{\star} = \dfrac{-p_{\star}'  + 2 \frac{\partial v_{\star}}{\partial y_{\star} }}{ Ra T_{\star}- Ra_{FS} }.
+# \end{equation}
+
+# First let's load the final state from the base case and calculate the expected dynamic topography with the code below.
+
+# + tags=["active-ipynb"]
+# # load velocity and pressure from final state of the base case run (we have saved earlier it earlier!).
```

Comment:
```suggestion
# # load velocity and pressure from final state of the base case run (we have saved it earlier!).
```

Does this also mean we don't need the `Final_State_base.h5` file committed?

Reply:
I think that is a yes. I do not have it on my branch, and that does not cause an issue.

Reply:
I think we do need the file because this is specifically for plotting a previous base case run to compare with the free surface run at steady state.

Reply:
Oh, then that may explain why I keep getting a run failure despite the test success! I will try adding it in to see.

Reply:
Can confirm everything ran successfully.